presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java

```diff
@@ -265,8 +266,18 @@ public static ConnectorPushdownFilterResult pushdownFilter(
 
         LogicalRowExpressions logicalRowExpressions = new LogicalRowExpressions(rowExpressionService.getDeterminismEvaluator(), functionResolution, functionMetadataManager);
         List<RowExpression> conjuncts = extractConjuncts(decomposedFilter.getRemainingExpression());
-        RowExpression dynamicFilterExpression = extractDynamicConjuncts(conjuncts, logicalRowExpressions);
-        RowExpression remainingExpression = extractStaticConjuncts(conjuncts, logicalRowExpressions);
+        ImmutableList.Builder<RowExpression> dynamicConjuncts = ImmutableList.builder();
+        ImmutableList.Builder<RowExpression> staticConjuncts = ImmutableList.builder();
+        for (RowExpression conjunct : conjuncts) {
+            if (isDynamicFilter(conjunct) || useDynamicFilter(conjunct, tableHandle, columnHandles)) {
```

Comment:
Lets have only one method in the if clause, which is the protected method. It should default to isDynamicFilter().

Reply:
I believe pushing this inside one method, might have complication.

for isDynamicFilter, we never want them to push down and that should be an invariant. Overriding the method then overrider need to enforce this constraint, that should not be upto the implementation. so I would prefer to keep this as it is.

Reply:
should we rename the dynamic conjuncts variable if it includes some non-dynamic conjuncts?

Reply:
I believe they are still dynamic, one is for dynamicFilter and other is dynamic columns. But if there are other names, I can change it.

Reply:
ah okay.  I think i just find isDynamicFilter() and useDynamicFilter() to be confusing. useDynamicFilter() sounds like it's not really a dynamic filter, but we're just using that field.  something like `canBeAppliedAtScan()`  might be clearer. not sure if it's too much of a mouthful.